src/config_map.c

```diff
@@ -0,0 +1,155 @@
+#include "config_map.h"
+
+#include "config.h"
+#include "gfx/common.h"
+#include "global/const.h"
+
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdint.h>
+
+static const CONFIG_OPTION_ENUM_MAP m_UIStyles[] = {
+    { "ps1", UI_STYLE_PS1 },
+    { "pc", UI_STYLE_PC },
+    { NULL, -1 },
+};
+
+static const CONFIG_OPTION_ENUM_MAP m_BarShowingModes[] = {
+    { "default", BSM_DEFAULT },
+    { "flashing-or-default", BSM_FLASHING_OR_DEFAULT },
+    { "flashing-only", BSM_FLASHING_ONLY },
+    { "always", BSM_ALWAYS },
+    { "never", BSM_NEVER },
+    { "ps1", BSM_PS1 },
+    { NULL, -1 },
+};
+
+static const CONFIG_OPTION_ENUM_MAP m_BarLocations[] = {
+    { "top-left", BL_TOP_LEFT },
+    { "top-center", BL_TOP_CENTER },
+    { "top-right", BL_TOP_RIGHT },
+    { "bottom-left", BL_BOTTOM_LEFT },
+    { "bottom-center", BL_BOTTOM_CENTER },
+    { "bottom-right", BL_BOTTOM_RIGHT },
+    { NULL, -1 },
+};
+
+static const CONFIG_OPTION_ENUM_MAP m_BarColors[] = {
+    { "gold", BC_GOLD },   { "blue", BC_BLUE },     { "grey", BC_GREY },
+    { "red", BC_RED },     { "silver", BC_SILVER }, { "green", BC_GREEN },
+    { "gold2", BC_GOLD2 }, { "blue2", BC_BLUE2 },   { "pink", BC_PINK },
+    { NULL, -1 },
+};
+
+static const CONFIG_OPTION_ENUM_MAP m_ScreenshotFormats[] = {
+    { "jpg", SCREENSHOT_FORMAT_JPEG },
+    { "jpeg", SCREENSHOT_FORMAT_JPEG },
+    { "png", SCREENSHOT_FORMAT_PNG },
+    { NULL, -1 },
+};
+
+const CONFIG_OPTION g_ConfigOptionMap[] = {
+    // clang-format off
+    { .name = "disable_healing_between_levels", .type = COT_BOOL,   .target = &g_Config.disable_healing_between_levels,      .default_value = &(bool){false}},
```

Comment:
There are some compiler warnings - I appreciate they're not relevant in these cases, but is it worth putting explicit `.param = NULL` here to reduce the noise?

`[9/165] Compiling C object TR1X.exe.p/src_config_map.c.obj
../../src/config_map.c:53:5: warning: missing initializer for field ‘param’ of ‘CONFIG_OPTION’ [-Wmissing-field-initializers]`

Reply:
Sorry, I forgot to compile it for Windows. I removed the warnings.